syntax = "proto3";

package llm;

service LLMService {
    // 文本生成
    rpc GenerateText (TextRequest) returns (TextResponse) {}
    // 获取模型信息
    rpc GetModelInfo (ModelInfoRequest) returns (ModelInfoResponse) {}
}

message TextRequest {
    string prompt = 1;
    string model_name = 2;  // deepseek-7b 或 qwen-7b
    optional int32 max_length = 3;
    optional float temperature = 4;
}

message TextResponse {
    string generated_text = 1;
    int32 tokens_generated = 2;
}

message ModelInfoRequest {
    string model_name = 1;
}

message ModelInfoResponse {
    string model_name = 1;
    int32 max_length = 2;
    string architecture = 3;
}